plugins/module_utils/compose_v2.py

```diff
@@ -736,6 +727,30 @@ def __init__(self, client, min_version=MINIMUM_COMPOSE_VERSION):
         # more precisely in https://github.com/docker/compose/pull/11478
         self.use_json_events = self.compose_version >= LooseVersion('2.29.0')
 
+    def get_compose_version(self):
+        return self.get_compose_version_from_cli() or self.get_compose_version_from_api()
+
+    def get_compose_version_from_cli(self):
+        rc, version_info, stderr = self.client.call_cli('compose', 'version', '--format', 'json')
+        if rc:
+            return None
+        try:
+            return json.loads(version_info)['version'].lstrip('v')
```

Comment:
One final thing, let's handle `dev` here as well:
```suggestion
            version = json.loads(version_info)['version']
            if version == 'dev':
                return None
            return version.lstrip('v')
```